src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,495 @@
+            ), "Either asset_name or input_data_asset_name must be provided"
+            capture_time = datetime_to_name_string(datetime.now())
+            capture_result_config.asset_name = (
+                f"{input_data_asset_name}"
+                f"_{capture_result_config.process_name}"
+                f"_{capture_time}"
+            )
+
+        if capture_result_config.mount is None:
+            capture_result_config.mount = capture_result_config.asset_name
+
+        computation_source = Sources.Computation(
+            id=computation_id,
+        )
+        source = Source(computation=computation_source)
+        create_data_asset_request = CreateDataAssetRequest(
+            name=capture_result_config.asset_name,
+            tags=capture_result_config.tags,
+            mount=capture_result_config.mount,
+            source=source,
+            custom_metadata=capture_result_config.custom_metadata,
+        )
+
+        reg_result_response = self.co_client.create_data_asset(
+            create_data_asset_request
+        )
+        registered_results_response_json = reg_result_response.json()
+
+        # TODO: This step intermittently breaks. Adding extra check to help
+        #  figure out why.
+        if registered_results_response_json.get("id") is None:
+            raise KeyError(
+                f"Something went wrong registering"
+                f" {capture_result_config.asset_name}. "
+                f"Response Status Code: {reg_result_response.status_code}. "
+                f"Response Message: {registered_results_response_json}"
+            )
+
+        results_data_asset_id = registered_results_response_json["id"]
+        response_res_available = self._wait_for_data_availability(
+            data_asset_id=results_data_asset_id
+        )
+
+        if response_res_available.status_code != 200:
+            raise FileNotFoundError(f"Unable to find: {results_data_asset_id}")
+
+        # Make captured results viewable to everyone
+        if capture_result_config.viewable_to_everyone:
+            update_res_perm_response = self.co_client.update_permissions(
+                data_asset_id=results_data_asset_id, everyone="viewer"
+            )
+            logger.info(
+                f"Updating permissions {update_res_perm_response.status_code}"
+            )
+        return reg_result_response
+
+    def process(self) -> dict:
+        """
+        Method to process data with a Code Ocean capsule/pipeline.
+        """
+        responses = dict()
+
+        # 1. register data assets (optional)
+        if self.job_config.register_config is not None:
+            logger.info("Registering data asset")
+            register_data_asset_response = (
+                self._register_data_and_update_permissions(
+                    self.job_config.register_config
+                )
+            )
+            self.job_config.run_capsule_config.data_assets.append(
```

Comment:
Mutating the job_config attribute seems fishy. It makes this method potentially have different behavior if run in succession.

Reply:
@jtyoung84 I removed all in-place modifications of the configs